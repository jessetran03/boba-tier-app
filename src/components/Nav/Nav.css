nav {
  display: flex;
  padding: 0.5% 0;
  margin-bottom: 20px;
  background-color: rgb(255, 255, 255);
  box-shadow: 0 3px 3px #c7c7c7;
  justify-content: space-between;
  overflow: hidden;
  position: fixed;
  top: 0;
  width: 100%;
}

nav p {
  font-size: 17.5px;
}

nav select {
  padding: 5px;
  border-radius: 20px;
  width: 130px;
}

.nav-header img {
  max-width: 150px;
  height: auto;
  margin: 20px;
}

.nav-header, .nav-links {
  margin: auto 5px;
}

.active p {
  color: rgb(47, 173, 159);
}

.bars {
  margin: 0 20px;
  font-size: 45px;
}

.bars:hover {
  cursor: pointer;
}

.desktop-nav {
  display: none;
}

@media all and (min-width: 1200px) {
  .mobile-nav {
    display: none;
  }

  .nav-header {
    margin-left: 14%;
  }

  .nav-links {
    margin-right: 14%;
  }

  .desktop-nav {
    display: flex;
  }

  .desktop-nav p, .desktop-nav select {
    display: inline;
    margin-right: 30px;
  }
}