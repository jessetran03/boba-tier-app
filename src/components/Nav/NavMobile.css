.mobile-nav-links {
  position: fixed;
  margin: 0;
  padding: 0;
  width: 100%;
  top: 84px;
  left: 0px;
  background-color: rgb(47, 173, 159, 0.97);
}

.mobile-nav-links p {
  padding: 20px 0;
  font-size: 30px;
  font-weight: 600;
  transition: color 0.2s ease-out, background-color 0.2s ease-in-out;
}

.mobile-nav-links p:hover {
  background-color: rgb(47, 173, 159, 0.97);
  color: white;
}

.mobile-nav-links .active p {
  color: white;
  background-color: rgb(83, 83, 83);
  font-weight: 600;
}

.mobile-nav-links .active p:hover {
  background-color: rgb(83, 83, 83);
}

.mobile-nav-links hr {
  margin: 0 100px;
  border: 1px solid rgb(216, 216, 216);
}