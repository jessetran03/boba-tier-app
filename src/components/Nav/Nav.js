import React, { Component } from 'react'
import { NavLink, Link } from 'react-router-dom'
import TokenService from '../../services/token-service'
// import config from '../config'
import './Nav.css'

export default class Nav extends Component {
  state = {
    currentCity: ''
  }

  handleLogoutClick = () => {
    TokenService.clearAuthToken()
  }

  renderLoggedIn() {
    return (
      <>
          <NavLink exact to='/' activeClassName='active'>
            <p>Home</p>
          </NavLink>
          <NavLink to='/lists' activeClassName='active'>
            <p>My Lists</p>
          </NavLink>
          <NavLink to='/shops' activeClassName='active'>
            <p>Shop List</p>
          </NavLink>
          <NavLink to='/rankings' activeClassName='active'>
            <p>Rankings</p>
          </NavLink>
          <NavLink to='/shops/2' activeClassName='active'>
            <p>Sharetea</p>
          </NavLink>
          <select>
            <option value='Houston'>Houston</option>
            <option value='Austin'>Austin</option>
            <option value='Dallas'>Dallas</option>
            <option value='San Francisco'>San Francisco</option>
          </select>
          <Link onClick={this.handleLogoutClick} to='/'>
            <p>Log Out</p>
          </Link>
      </>
    )
  }

  renderLoggedOut() {
    return (
      <>
          <NavLink exact to='/' activeClassName='active'>
            <p>Home</p>
          </NavLink>
          <NavLink to='/shops' activeClassName='active'>
            <p>Shop List</p>
          </NavLink>
          <NavLink to='/rankings' activeClassName='active'>
            <p>Rankings</p>
          </NavLink>
          <NavLink to='/shops/2' activeClassName='active'>
            <p>Sharetea</p>
          </NavLink>
          <select>
            <option value='Houston'>Houston</option>
          </select>
          <NavLink to='/login' activeClassName='active'>
            <p>Log In</p>
          </NavLink>
          <NavLink to='/register' activeClassName='active'>
            <p>Sign Up</p>
          </NavLink>
      </>
    )
  }

  render() {
    return (
      <nav>
        <section className='nav-header'>
          <Link to='/'>
            <h2>Boba Tier</h2>
          </Link>
        </section>
        <section className='nav-links'>
          {TokenService.hasAuthToken()
            ? this.renderLoggedIn()
            : this.renderLoggedOut()
          }
        </section>
      </nav>
    )
  }
}