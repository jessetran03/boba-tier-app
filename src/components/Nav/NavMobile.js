import React, { Component } from 'react'
import { NavLink, Link } from 'react-router-dom'
import TokenService from '../../services/token-service'
// import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
// import config from '../config'
import './NavMobile.css'

export default class NavMobile extends Component {

  componentDidMount() {
    document.addEventListener('mousedown', this.handleClick, false);
  }

  componentWillUnmount() {
    document.removeEventListener('mousedown', this.handleClick, false);
  }

  handleClick = (e) => {
    if (this.node.contains(e.target)) {
      return;
    }
    this.props.onHandleNav();
  }

  handleNavClick = () => {
    this.props.onHandleNav();
  }

  handleLogoutClick = () => {
    this.props.onHandleLogout()
    this.props.onHandleNav()
  }

  renderLoggedIn() {
    return (
      <>
          <NavLink exact to='/' onClick={this.handleNavClick} activeClassName='active'>
            <p>Home</p>
          </NavLink>
          <hr />
          <NavLink to='/shops' onClick={this.handleNavClick} activeClassName='active'>
            <p>Shop List</p>
          </NavLink>
          <hr />
          <NavLink to='/rankings' onClick={this.handleNavClick} activeClassName='active'>
            <p>Top 10 Drinks</p>
          </NavLink>
          <hr />
          <NavLink to='/lists' onClick={this.handleNavClick} activeClassName='active'>
            <p>My Ratings</p>
          </NavLink>
          <hr />
          <Link onClick={this.handleLogoutClick} to='/' >
            <p>Log Out</p>
          </Link>
      </>
    )
  }

  renderLoggedOut() {
    return (
      <>
        <NavLink exact to='/' onClick={this.handleNavClick} activeClassName='active' >
          <p>Home</p>
        </NavLink>
        <hr />
        <NavLink to='/shops' onClick={this.handleNavClick} activeClassName='active'>
          <p>Shop List</p>
        </NavLink>
        <hr />
        <NavLink to='/rankings' onClick={this.handleNavClick} activeClassName='active'>
          <p>Top 10 Drinks</p>
        </NavLink>
        <hr />
        <NavLink to='/login' onClick={this.handleNavClick} activeClassName='active'>
          <p>Log In</p>
        </NavLink>
        <hr />
        <NavLink to='/register' onClick={this.handleNavClick} activeClassName='active'>
          <p>Sign Up</p>
        </NavLink>
      </>
    )
  }

  render() {
    return (
      <section ref={node => this.node = node} className='mobile-nav-links mobile-nav'>
        {TokenService.hasAuthToken()
          ? this.renderLoggedIn()
          : this.renderLoggedOut()
        }
      </section>
    )
  }
}