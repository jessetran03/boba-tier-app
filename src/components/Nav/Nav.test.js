import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import Nav from './Nav'
import renderer from 'react-test-renderer'

it('renders Nav without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <BrowserRouter>
      <Nav />
    </BrowserRouter>,
    div
  )
  ReactDOM.unmountComponentAtNode(div)
})

it('render Nav without data', () => {
  const wrapper = renderer.create(
    <BrowserRouter>
      <Nav />
    </BrowserRouter>
  )
    .toJSON();
  expect(wrapper).toMatchSnapshot()
})