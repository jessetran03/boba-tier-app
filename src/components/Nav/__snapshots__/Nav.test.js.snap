// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Nav without data 1`] = `
<nav>
  <section
    className="nav-header"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Boba Tier Logo"
        src="boba-tier-logo.png"
      />
    </a>
  </section>
  <section
    className="nav-links"
  >
    <section
      className="desktop-nav"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <p>
          Home
        </p>
      </a>
      <a
        aria-current={null}
        href="/shops"
        onClick={[Function]}
      >
        <p>
          Shop List
        </p>
      </a>
      <a
        aria-current={null}
        href="/rankings"
        onClick={[Function]}
      >
        <p>
          Top 10
        </p>
      </a>
      <select>
        <option
          value="Houston"
        >
          Houston, TX
        </option>
      </select>
      <a
        aria-current={null}
        href="/login"
        onClick={[Function]}
      >
        <p>
          Log In
        </p>
      </a>
      <a
        aria-current={null}
        href="/register"
        onClick={[Function]}
      >
        <p>
          Sign Up
        </p>
      </a>
    </section>
    <section
      className="mobile-nav"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-bars fa-w-14 bars"
        data-icon="bars"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </section>
  </section>
</nav>
`;
